virtual-programs/receipt-printer.folk

```diff
@@ -0,0 +1,95 @@
+namespace eval EscPos {
+    proc printProgram {printer id code} {
+        if {[file exists "$::env(HOME)/folk-printed-programs/$id.folk"]} {
+            error "Program $id already exists on disk. Aborting print."
+        }
+        writeFolkFile $id $code
+        writeGeomFile $printer $id
+
+        set matches [Statements::findMatches [list /someone/ claims $printer is at /address/]]
+        set matchDict [lindex $matches 0]
+        if {![dict exists $matchDict address]} { return }
+        set address [dict get $matchDict address]
+
+        set printerSocket [socket $address 9100]
+
+        fconfigure $printerSocket -translation binary -buffering none
+        set template {
+            [init]
+            [tag $id]
+            [feed 1]
+            $id ([clock format [clock seconds] -timezone :America/Denver -format "%a, %d %b %Y, %r"])
+            [feed 2]
+            $code
+            [feed 3]
+            [cut]
+        }
+        puts -nonewline $printerSocket [subst [prepareTemplate $template]]
+        close $printerSocket
+    }
+
+    proc prepareTemplate {template} {
+        set trimmed [lmap line [split $template "\n"] { string trim $line }]
+        set singleLine [join $trimmed ""]
+        return $singleLine
```

Comment:
@osnr I tried to find a way to `return [uplevel "subst $singleLine"]` here but it kept acting weird so I just call the subst on line 27 🤷 

I guess I need to hone my Tcl skills some more.

Reply:
Maybe `[uplevel [list subst $singleLine]]`?